### PR TITLE
Host progression

### PR DESCRIPTION
**Pull request by Elizabeth-Hayman**

### CODE REVIEW

File: pyEpiabm/pyEpiabm/tests/test_host_progression_sweep.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+class TestHostProgressionSweep(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        '''Sets up a population we can use throughout the test.
+        2 people in one microcell
+        '''
+        cls.test_population = pe.Population()
+        cls.test_population.add_cells(1)
+        cls.test_cell = cls.test_population.cells[0]
```
Comment (KCGallagher) [inline pyEpiabm/pyEpiabm/tests/test_host_progression_sweep.py:14]: Is this (`test_cell`) used anywhere else?

**Review by KCGallagher — CHANGES_REQUESTED** [review]

File: pyEpiabm/pyEpiabm/tests/test_host_progression_sweep.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import unittest
+import pyEpiabm as pe
+
+
+class TestHostProgressionSweep(unittest.TestCase):
+    '''Test the Host Progression Sweep function.
```
Comment (rcw5890) [inline pyEpiabm/pyEpiabm/tests/test_host_progression_sweep.py:6]:
```suggestion
    """Test the Host Progression Sweep function.
```
and so forth

File: pyEpiabm/pyEpiabm/host_progression_sweep.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    def __call__(self, time: int):
+        """Method that sweeps through all people in the population
+        and updates their infection status if it is time and assigns
+        them a next infection status and a new time of next status
+        change.
+
+        : param time: Current simulation time in days
+        : type time: float
+        """
+        for cell in self._population.cells:
```
Comment (rcw5890) [inline pyEpiabm/pyEpiabm/host_progression_sweep.py:44]: Should we check that t is int ? to avoid problems if they call `__call__(None)`

**Review by rcw5890 — CHANGES_REQUESTED** [review]

File: pyEpiabm/pyEpiabm/host_progression_sweep.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+class HostProgressionSweep(AbstractSweep):
+    '''Class for sweeping through population and updating host infection status
+    and time to next infection status change.
+    '''
+
+    def _update_time_to_status_change(self):
+        """Method that assigns time until next infection status update,
+         given as a random integer between 1 and 10.
+        """
```
Comment (KCGallagher) [inline pyEpiabm/pyEpiabm/host_progression_sweep.py:14]:
```suggestion
        
        :return: Time until next infection status update
        :rtype: int
        """
```

**Review by KCGallagher — CHANGES_REQUESTED** [review]

File: pyEpiabm/pyEpiabm/tests/test_host_progression_sweep.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    '''
```
Comment (KCGallagher) [inline pyEpiabm/pyEpiabm/tests/test_host_progression_sweep.py:7]:
```suggestion
    """
```

**Review by KCGallagher — CHANGES_REQUESTED** [review]

File: pyEpiabm/pyEpiabm/host_progression_sweep.py (hunk 4 above)
Comment (KCGallagher) [reply pyEpiabm/pyEpiabm/host_progression_sweep.py:14]: @lukedtc are you sure this is resolved? Think this change should be committed but let me know if you have any issues with it!

**Review by KCGallagher — CHANGES_REQUESTED** [review]

**Review by KCGallagher — APPROVED** [review]

**Review by rcw5890 — APPROVED** [review]

**Pull request merged**